lib/dal/src/func/binding/management.rs

```diff
@@ -70,93 +69,134 @@ impl ManagementBinding {
+                    // Since it's already installed, we don't include it in "installable_schemas"
+                    installable_schemas.remove(schema.name());
+
                     let variant_id =
-                        ManagementPrototype::get_schema_variant_id(ctx, prototype_id).await?;
+                        Schema::get_or_install_default_variant(ctx, schema.id()).await?;
+
                     let root_prop = Prop::get_by_id(
                         ctx,
                         SchemaVariant::get_root_prop_id(ctx, variant_id).await?,
                     )
                     .await?;
 
-                    let (_, reverse_map) = prototype.managed_schemas_map(ctx).await?;
-
-                    let mut component_create_types = vec![];
-                    let mut component_update_types = vec![];
-                    for (schema_id, name) in reverse_map {
-                        let variant_id =
-                            Schema::get_or_install_default_variant(ctx, schema_id).await?;
+                    let sv_type = root_prop.ts_type(ctx).await?;
+
+                    let json_name = serde_json::to_string(schema.name())?;
+
+                    let component_input_type = format!(
+                        r#"
+                            {{
+                                kind: {json_name},
+                                properties?: {sv_type},
+                                geometry?: {{ [key: string]: Geometry }},
+                                connect?: {{
+                                    from: string,
+                                    to: {{
+                                        component: string;
+                                        socket: string;
+                                    }}
+                                }}[],
+                                parent?: string,
+                            }}
+                        "#
+                    );
+
+                    let component_create_type = format!(
+                        r#"
+                            {{
+                                kind?: {json_name},
+                                properties?: {sv_type},
+                                geometry?: Geometry | {{ [key: string]: Geometry }},
+                                connect?: Connection[],
+                                parent?: string,
+                            }}
+                        "#
+                    );
+                    component_create_types.push(component_create_type);
+                    component_input_types.push(component_input_type);
+                }
 
-                        let root_prop = Prop::get_by_id(
-                            ctx,
-                            SchemaVariant::get_root_prop_id(ctx, variant_id).await?,
-                        )
-                        .await?;
+                // Add a generic type for any uninstalled modules, but with the names so that
```

Comment:
After producing types for installed schemas, we produce a single create and input type that covers all uninstalled schemas; `kind` becomes e.g. `kind: "AWS::EC2::VPC" | "AWS::EC2::Subnet" | ...,`